### PR TITLE
Refactor core

### PR DESCRIPTION
**Pull request by xiao-lix**
move DiagnosticLogger, _InternalLogMessage, Notification to a wrapper class from core

### CODE REVIEW

File: extensions/applicationinsights-analytics-js/Tests/ApplicationInsights.tests.ts [hunk 1]
```diff
@@ -293,7 +293,7 @@ export class ApplicationInsightsTests extends TestClass {
                     [plugin]
                 );
                 var appInsights = new ApplicationInsights();
-                appInsights.initialize({instrumentationKey: core.config.instrumentationKey}, core, []);
+                appInsights.initialize({instrumentationKey: core.baseCore.config.instrumentationKey}, core, []);
```
Comment (hectorhdzg) [inline extensions/applicationinsights-analytics-js/Tests/ApplicationInsights.tests.ts]: config must be the same between core and baseCore, please test this using core.config.instrumentationKey

File: shared/AppInsightsCore/src/JavaScriptSDK.Tests/Selenium/ApplicationInsightsCore.Tests.ts [hunk 2]
```diff
@@ -190,11 +190,11 @@ export class ApplicationInsightsCoreTests extends TestClass {
                 const channelPlugin = new ChannelPlugin();
                 const appInsightsCore = new AppInsightsCore();
                 appInsightsCore.initialize({ instrumentationKey: expectedIKey }, [channelPlugin]);
-                const validateStub = this.sandbox.stub(appInsightsCore, "_validateTelmetryItem");
+                const validateStub = this.sandbox.stub(appInsightsCore.baseCore, "_validateTelmetryItem");
 
                 // Act
                 const bareItem: ITelemetryItem = { name: 'test item' };
-                appInsightsCore.track(bareItem);
+                appInsightsCore.baseCore.track(bareItem);
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/src/JavaScriptSDK.Tests/Selenium/ApplicationInsightsCore.Tests.ts]: Call appInsightsCore.track here, this is main scenario and we want to be sure it works fine

File: shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts [hunk 3]
```diff
@@ -190,50 +191,18 @@ export class BaseCore implements IAppInsightsCore {
...
-
     private _validateTelmetryItem(telemetryItem: ITelemetryItem) {
 
         if (CoreUtils.isNullOrUndefined(telemetryItem.name)) {
-            this._notifyInvalidEvent(telemetryItem);
             throw Error("telemetry name required");
         }
 
         if (CoreUtils.isNullOrUndefined(telemetryItem.time)) {
-            this._notifyInvalidEvent(telemetryItem);
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts]: You need to notify if the event is dropped here, move this validation to AICore where notification manager is available

File: shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts [hunk 4]
```diff
@@ -175,6 +172,10 @@ export class BaseCore implements IAppInsightsCore {
         // do basic validation before sending it through the pipeline
         this._validateTelmetryItem(telemetryItem);
 
+        this.trackItem(telemetryItem);
+    }
+
+    trackItem(telemetryItem: ITelemetryItem) {
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts]: I don't understand why you need another method here

File: shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts [hunk 5]
```diff
@@ -37,7 +39,29 @@ export class AppInsightsCore implements IAppInsightsCore {
...
 
     track(telemetryItem: ITelemetryItem) {
-        this.baseCore.track(telemetryItem);
+        if (telemetryItem === null) {
+            this._notifyInvalidEvent(telemetryItem);
+            // throw error
+            throw Error("Invalid telemetry item");
+        }
+
+        if (!telemetryItem.iKey) {
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts]: These 3 checks need to be in baseCore

File: shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+import { CoreUtils } from "./CoreUtils";
+import { INotificationManager } from '../JavaScriptSDK.Interfaces/INotificationManager';
+import { IDiagnosticLogger } from "../JavaScriptSDK.Interfaces/IDiagnosticLogger";
+import { ChannelController } from './ChannelController';
+
+"use strict";
+
+const validationError = "Extensions must provide callback to initialize";
+
+export class BaseCore implements IAppInsightsCore {
```
Comment (hectorhdzg) [inline shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts]: Add this class export in applicationinsights-core-js.ts so it can be used in other packages

Comment (xiao-lix) [reply shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts]: yeah this was added already 😁

**Pull request merged**